src/obfuscator.hpp

```diff
@@ -31,8 +31,8 @@ class obfuscator {
     static constexpr std::string_view default_key_regex_str{
         R"((p(ass)?w(or)?d|pass(_?phrase)?|secret|(api_?|private_?|public_?)key)|token|consumer_?(id|key|secret)|sign(ed|ature)|bearer|authorization)"};
 
-    std::unique_ptr<re2::RE2> key_regex{nullptr};
-    std::unique_ptr<re2::RE2> value_regex{nullptr};
+    std::shared_ptr<re2::RE2> key_regex{nullptr};
+    std::shared_ptr<re2::RE2> value_regex{nullptr};
```

Comment:
Same as mentioned before. It seems that would should be shared would be the obfuscator, rather than relying on the copy constructor to create a new obfuscator with shared ownership of the internal state.

Reply:
Done